app/controllers/order_steps_controller.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+class OrderStepsController < ApplicationController
+  include Wicked::Wizard
+
+  before_action :go_root_unless_order
+  before_action :initialize_form
+
+  steps :fast_sign, :address, :delivery, :payment, :confirm, :complete
+
+  def show
+    case step
+    when :fast_sign then next_step and return if current_user
+    when :address, :delivery, :payment
+      step_to :fast_sign and return unless current_user
+      step_to :complete and return if last_active_order.in_confirmation?
+      step_to :confirm and return if last_active_order.filled? && !order_params[:editing].present?
+    when :confirm then return check_info_steps
+    when :complete
+      step_to :fast_sign and return unless current_user
+      step_to :confirm and return unless last_active_order.in_confirmation?
+    end
+    render_wizard
```

Comment:
It is really complex method for controller. too many if. Can we move it somewhere and divide?